lib/barcelona/chaos.rb

```diff
@@ -0,0 +1,42 @@
+module Barcelona
+  class Chaos
+    attr_accessor :district, :terminate_count
+
+    def self.run(district_names, count:)
+      District.where(name: district_names.uniq).each do |d|
+        self.new(d, count).run
+      end
+    end
+
+    def initialize(district, terminate_count)
+      @district = district
+      @terminate_count = terminate_count
+    end
+
+    def run
+      instances = district.container_instances.sort_by { |i| i[:launch_time] }
+      non_active_count = instances.select { |i| i[:status] != 'ACTIVE' }.count
+      if non_active_count > 0
+        notify("#{district.name} has one or more non-active instance(s). Skipping", level: :warn)
+        return
+      end
```

Comment:
Do you need a count here? If all you care about is that there is *any* match, think this should work:

```ruby
if instances.detect { |i| i[:status] != 'ACTIVE' }
  ...
end
```